mm2src/mm2_io/src/fs.rs

```diff
@@ -192,19 +192,50 @@ where
     json::from_slice(&content).map_to_mm(FsJsonError::Deserializing)
 }
 
-/// Read the `dir_path` entries trying to deserialize each as the `T` type.
+async fn filter_files_with_extension(dir_path: &Path, extension: &str) -> IoResult<Vec<PathBuf>> {
+    let ext = Some(OsStr::new(extension).to_ascii_lowercase());
+    let entries = read_dir_async(dir_path)
+        .await?
+        .into_iter()
+        .filter(|path| path.extension().map(|ext| ext.to_ascii_lowercase()) == ext)
+        .collect();
+    Ok(entries)
+}
+
+/// Helper function to extract file names or stems based on the provided extraction function.
+fn extract_file_identifiers<F>(entries: Vec<PathBuf>, extractor: F) -> Vec<String>
+where
+    F: Fn(&Path) -> Option<&OsStr>,
+{
+    entries
+        .into_iter()
+        .filter_map(|path| extractor(&path).and_then(OsStr::to_str).map(ToOwned::to_owned))
+        .collect()
+}
+
+/// Reads the file names with the specified extension from the given directory path.
+pub async fn read_file_names_with_extension(dir_path: &Path, extension: &str) -> IoResult<Vec<String>> {
+    let entries = filter_files_with_extension(dir_path, extension).await?;
+    Ok(extract_file_identifiers(entries, Path::file_name))
+}
+
+/// Reads the file stems with the specified extension from the given directory path.
+/// The stem is the file name without the extension.
+pub async fn read_file_stems_with_extension(dir_path: &Path, extension: &str) -> IoResult<Vec<String>> {
+    let entries = filter_files_with_extension(dir_path, extension).await?;
+    Ok(extract_file_identifiers(entries, Path::file_stem))
+}
```

Comment:
nit: I would take the extractor as an arg to not duplicate this function.

Reply:
I think it will be harder for others to use these functions if they had to pass `Path::file_name` or `Path::file_stem` to them. These are public functions that can be used outside this crate and users of the function shouldn't look for which extractor to use.

Reply:
Can a bool flag (something like `ignore_file_extension`) help in that case?

Reply:
> Can a bool flag (something like ignore_file_extension) help in that case?

Yes, thanks for the suggestion. Used `include_extension` bool flag here https://github.com/KomodoPlatform/komodo-defi-framework/pull/2202/commits/bea5479fdb189b278c12911d8be284303ef2973a